src/lib/crypto/hash.cpp

```diff
@@ -233,13 +255,24 @@ pgp_hash_finish(pgp_hash_t *hash, uint8_t *out)
         return 0;
     }
 
-    if (out && botan_hash_final((botan_hash_t) hash->handle, out)) {
+    Botan::HashFunction *hash_fn = static_cast<Botan::HashFunction *>(hash->handle);
+    hash->handle = NULL;
+    if (hash_fn) {
+        try {
+            if (out)
+                hash_fn->final(out);
```

Comment:
By the way, style nit here, we always use braces even on one-line conditionals. See the [Style Guide](https://github.com/rnpgp/rnp/blob/master/docs/develop.adoc#style-guide).

Not something to hold up a merge though.